skcosmo/selection/CUR.py

```diff
@@ -0,0 +1,297 @@
+from abc import abstractmethod
+import numpy as np
+from scipy.sparse.linalg import eigs as speig
+
+from skcosmo.pcovr.pcovr_distances import get_Ct, get_Kt
+from .orthogonalizers import feature_orthogonalizer, sample_orthogonalizer
+
+
+class _CUR:
+    """
+    Base class for CUR selection methods
+    Requires a product, typically the gram or covariance matrix, \
+    from which to compute the importance score
+
+    If the model is iterative, the orthogonalize method must be overwritten
+    to orthogonalize the input matrices after each iteration.
+
+
+    :param iterative: whether to orthogonalize after each selection,
+                      defaults to `true`
+    :type iterative: boolean
+
+    :param k: number of eigenvectors to compute the importance score with,
+              defaults to 1
+    :type k: int
+
+    :param tolerance: threshold below which values will be considered 0,
+                      defaults to 1E-12
+    :type tolerance: float0
+    """
+
+    def __init__(self, iterative=True, tolerance=1e-12, k=1, **kwargs):
+
+        self.k = k
+        self.iter = iterative
+        self.tol = tolerance
+        self.idx = []
+        self.pi = []
+
+    def select(self, n):
+        """
+        Method for CUR select based upon a product of the input matrices
+
+        :param n: number of selections to make
+        :type n: int
+
+        :returns: list of selected indices
+        :rtype: list of int
+
+        """
+
+        if len(self.idx) > n:
+            return self.idx[:n]
+
+        for i in range(len(self.idx), n):
+            # try:
+            if self.iter:
+                v, U = speig(self.product, k=self.k, tol=self.tol)
+                U = U[:, np.flip(np.argsort(v))]
+                pi = (np.real(U)[:, : self.k] ** 2.0).sum(axis=1)
+            pi[self.idx] = 0.0
+            self.idx.append(pi.argmax())
+            self.pi.append(max(pi))
+
+            self.orthogonalize()
+            self.product = self.get_product()
+
+            if np.isnan(self.product).any():
+                print(
+                    f"The product matrix has rank {i}. "
+                    + f"n_select reduced from {n} to {i}."
+                )
+                return self.idx
+
+        return self.idx
+
+    @abstractmethod
+    def get_product(self):
+        """
+        Abstract method for computing the inner or outer product of the
+        input matrices
+        """
+        return
+
+    @abstractmethod
+    def orthogonalize(self):
+        """Method to orthogonalize matrices"""
+        return
+
+
+class sampleCUR(_CUR):
```

Comment:
Python uses CamelCase for class name

```suggestion
class SampleCUR(_CUR):
```